### PR TITLE
Fixed a few find all refs bugs on values

### PR DESCRIPTION
**Pull request by TIHan**
Since we changed the way we do find all references in VS, we had a few regressions. This should fix the ones mentioned here: https://github.com/dotnet/fsharp/issues/9863

For context, we risked regressions due to the significant memory reduction that came with the change. I'm sure there are more fixes to do; they should be relatively straight forward though.

The one thing with find all references that is not going to be straight forward is performing a find all references on a currently editing buffer (a file that is not saved yet). Unfortunately, a currently editing file is not consolidated to the way type checking works on a project as a whole; it's handled separately. We also have two different ways of getting symbol information, and that's determined if a construct is public or private.

One way to solve this is probably just implementing the FileSystem shim to forward to the editing buffer in VS, which is a giant hack. At some point, we should figure out how to make the F# compiler not rely on a file system; it should work directly with streams and/or buffers.

### CODE REVIEW

**Comment by cartermp** [comment]
fixes https://github.com/dotnet/fsharp/issues/9863, https://github.com/dotnet/fsharp/issues/9750, https://github.com/dotnet/fsharp/issues/8823

Note that the following are still issues with this PR:

* https://github.com/dotnet/fsharp/issues/9275
* https://github.com/dotnet/fsharp/issues/9924
* https://github.com/dotnet/fsharp/issues/6243#issuecomment-673734020 (see comment)

**Comment by cartermp** [comment]
@TIHan I'm fine with taking this as-is for now since it does fix 3 distinct bugs. But there are 2 it still doesn't solve and an inconsistency in another. What are your thoughts?

File: src/fsharp/service/ItemKey.fs [hunk 1]
```diff
@@ -228,71 +222,67 @@ and [<Sealed>] ItemKeyStoreBuilder() =
                 writeILType x)
             writeILType mref.ReturnType
 
-    let rec writeType (ty: TType) =
-        match ty with
+    let rec writeType isStandalone (ty: TType) =
```
Comment (cartermp) [inline src/fsharp/service/ItemKey.fs:225]: What does `isStandalone` refer to?

Comment (TIHan) [reply src/fsharp/service/ItemKey.fs:225]: Referring to if a type is not nested as part of a symbol.

```
List<int> // when writing the key for `List<_>`, we say that type argument 'int' is not standalone
               // when writing the key for 'int', we say the type is standalone
```

**Comment by TIHan** [comment]
We should take the fix and keep iterating on the others.

**Pull request merged**